src/lib/fetch.ts

```diff
@@ -105,13 +145,35 @@ export default function fetch(url: string, options: RequestInit = { headers: new
     headers.set('x-github-run-id', process.env.GITHUB_RUN_ID);
     headers.set('x-github-run-number', process.env.GITHUB_RUN_NUMBER);
     headers.set('x-github-sha', process.env.GITHUB_SHA);
+
+    if (filePath) {
+      /**
+       * Constructs a full URL to the file using GitHub Actions runner variables
+       * @see {@link https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables}
+       * @example https://github.com/readmeio/rdme/blob/cb4129d5c7b51ff3b50f933a9c7d0c3d0d33d62c/documentation/rdme.md
+       */
+      headers.set(
+        'x-readme-source-url',
+        `${process.env.GITHUB_SERVER_URL}/${process.env.GITHUB_REPOSITORY}/blob/${process.env.GITHUB_SHA}/${filePath}`
+      );
+    }
+  }
+
+  if (isCI()) {
+    headers.set('x-rdme-ci', ciName());
   }
 
   headers.set('x-readme-source', source);
 
+  if (filePath && fileOpts.fileType === 'url') {
+    headers.set('x-readme-source-url', filePath);
```

Comment:
Do we only want to log these in GH like we're doing for `path`?

Reply:
I think if someone is running a command in any environment (CI, local script, etc.) and their spec is pointed to a URL, I think it's safe to log that URL as the source URL right? Like if someone were to do something like this:

```sh
rdme openapi https://github.com/readmeio/oas-examples/blob/main/3.0/json/petstore.json
```